geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/ProtobufSerializationService.java

```diff
@@ -36,58 +37,59 @@ public ProtobufSerializationService() {}
    */
   @Override
   public BasicTypes.EncodedValue encode(Object value) throws EncodingException {
-    if (value == null) {
-      return BasicTypes.EncodedValue.getDefaultInstance();
-    }
-
     BasicTypes.EncodedValue.Builder builder = BasicTypes.EncodedValue.newBuilder();
 
-    try {
-      ProtobufEncodingTypes protobufEncodingTypes = ProtobufEncodingTypes.valueOf(value.getClass());
-      switch (protobufEncodingTypes) {
-        case INT: {
-          builder.setIntResult((Integer) value);
-          break;
-        }
-        case LONG: {
-          builder.setLongResult((Long) value);
-          break;
-        }
-        case SHORT: {
-          builder.setShortResult((Short) value);
-          break;
-        }
-        case BYTE: {
-          builder.setByteResult((Byte) value);
-          break;
-        }
-        case DOUBLE: {
-          builder.setDoubleResult((Double) value);
-          break;
-        }
-        case FLOAT: {
-          builder.setFloatResult((Float) value);
-          break;
-        }
-        case BINARY: {
-          builder.setBinaryResult(ByteString.copyFrom((byte[]) value));
-          break;
-        }
-        case BOOLEAN: {
-          builder.setBooleanResult((Boolean) value);
-          break;
-        }
-        case STRING: {
-          builder.setStringResult((String) value);
-          break;
-        }
-        case PDX_OBJECT: {
-          builder.setJsonObjectResult(jsonPdxConverter.encode((PdxInstance) value));
-          break;
+    if (value == null) {
+      builder.setNullResult(NullValue.NULL_VALUE);
+    } else {
+      try {
+        ProtobufEncodingTypes protobufEncodingTypes =
+            ProtobufEncodingTypes.valueOf(value.getClass());
+        switch (protobufEncodingTypes) {
+          case INT: {
+            builder.setIntResult((Integer) value);
+            break;
+          }
+          case LONG: {
+            builder.setLongResult((Long) value);
+            break;
+          }
+          case SHORT: {
+            builder.setShortResult((Short) value);
+            break;
+          }
+          case BYTE: {
+            builder.setByteResult((Byte) value);
+            break;
+          }
+          case DOUBLE: {
+            builder.setDoubleResult((Double) value);
+            break;
+          }
+          case FLOAT: {
+            builder.setFloatResult((Float) value);
+            break;
+          }
+          case BINARY: {
+            builder.setBinaryResult(ByteString.copyFrom((byte[]) value));
+            break;
+          }
+          case BOOLEAN: {
+            builder.setBooleanResult((Boolean) value);
+            break;
+          }
+          case STRING: {
+            builder.setStringResult((String) value);
+            break;
+          }
+          case PDX_OBJECT: {
+            builder.setJsonObjectResult(jsonPdxConverter.encode((PdxInstance) value));
+            break;
+          }
```

Comment:
This isn't due to your change, but should we have a default: in this switch statement to throw an exception if we missed a type?

Reply:
Oh wow, apparently java have an implicit noop default.  I guess I was expecting a implicit throwing default, which doesn't necessarily make more sense, but yeah, I'll add this as explicit case.